src/main/java/me/contaria/seedqueue/mixin/compat/sodium/ChunkBuilderMixin.java

```diff
@@ -33,23 +32,33 @@ public abstract class ChunkBuilderMixin {
     @Final
     private AtomicBoolean running;
 
-    @ModifyReturnValue(
-            method = "getMaxThreadCount",
-            at = @At("RETURN")
+    @Dynamic
+    @ModifyExpressionValue(
+            method = "startWorkers",
+            at = {
+                    // sodium 2.3.1 and below
+                    @At(value = "FIELD", target = "Lme/jellysquid/mods/sodium/client/render/chunk/compile/ChunkBuilder;limitThreads:I"),
+                    // sodium 2.4.0+
+                    @At(value = "FIELD", target = "Lme/jellysquid/mods/sodium/client/render/chunk/compile/ChunkBuilder;initialThreads:I")
+            },
+            require = 1, allow = 1
     )
```

Comment:
Realized from this that the old code was kind of buggy. This is also somewhat problematic (maybe? not quite sure how things work internally.)

For old sodium - this is good, I think. For new sodium we need to add another inject, I think, to override `hardLimitThreads` and set it to the same value. Otherwise, I believe that with this, we would have gradual thread creation within the wall, which is probably not desirable.